src/database/schema/ChannelListSchema.ts

```diff
@@ -323,6 +323,22 @@ class ChannelList implements BaseDbSchema {
     }
   }
 
+  static async getAllUnreadCount(db: SQLiteDatabase): Promise<number> {
+    try {
+      const selectQuery = `SELECT SUM(unread_count) as unread_count
+        FROM ${ChannelList.getTableName()}
+        WHERE expired_at IS NULL
+        OR datetime(expired_at) >= datetime('now')`;
+
+      const [results] = await db.executeSql(selectQuery);
+
+      return Promise.resolve(results?.rows?.raw()[0]?.unread_count || 0);
+    } catch (e) {
+      console.log(e);
+      return Promise.resolve(0);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new function `getAllUnreadCount` is generally well-written, but there's a potential issue with error handling. Currently, if an error occurs during the execution of the SQL query, it's logged to the console and the function returns 0. This could potentially mask issues in the future if the SQL query fails for some reason (e.g., due to a schema change). It might be better to let the error propagate up to the caller, or at least log more detailed information about the error.

```diff
-      console.log(e);
-      return Promise.resolve(0);
+      console.error('Error executing getAllUnreadCount:', e);
+      throw e;
```
This way, if something goes wrong, you'll have more information about what happened and can handle the error appropriately in the calling code.


<!-- This is an auto-generated comment by OSS CodeRabbit -->